code/datums/antagonists/datum_devil.dm

```diff
@@ -277,73 +252,74 @@ var/global/list/lawlorify = list (
-	for(var/X in owner.spell_list)
+/datum/antagonist/devil/proc/remove_spells()
+	for(var/X in owner.mind.spell_list)
 		var/obj/effect/proc_holder/spell/S = X
 		if(!istype(S, /obj/effect/proc_holder/spell/targeted/summon_contract) && !istype(S, /obj/effect/proc_holder/spell/targeted/conjure_item/violin))
-			owner.RemoveSpell(S)
+			owner.mind.RemoveSpell(S)
 
-/datum/devilinfo/proc/give_summon_contract()
-	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/summon_contract(null))
+/datum/antagonist/devil/proc/give_summon_contract()
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/summon_contract(null))
 
 
-/datum/devilinfo/proc/give_base_spells(give_summon_contract = 0)
+/datum/antagonist/devil/proc/give_base_spells(give_summon_contract = 0)
 	remove_spells()
-	owner.AddSpell(new /obj/effect/proc_holder/spell/fireball/hellish(null))
-	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/summon_pitchfork(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/fireball/hellish(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/summon_pitchfork(null))
 	if(give_summon_contract)
 		give_summon_contract()
 		if (obligation == OBLIGATION_FIDDLE)
-			owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/violin(null))
+			owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/violin(null))
 
-/datum/devilinfo/proc/give_lizard_spells()
+/datum/antagonist/devil/proc/give_lizard_spells()
 	remove_spells()
-	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/summon_pitchfork(null))
-	owner.AddSpell(new /obj/effect/proc_holder/spell/fireball/hellish(null))
-	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/infernal_jaunt(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/summon_pitchfork(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/fireball/hellish(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/infernal_jaunt(null))
 
-/datum/devilinfo/proc/give_true_spells()
+/datum/antagonist/devil/proc/give_true_spells()
 	remove_spells()
-	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/summon_pitchfork/greater(null))
-	owner.AddSpell(new /obj/effect/proc_holder/spell/fireball/hellish(null))
-	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/infernal_jaunt(null))
-	owner.AddSpell(new /obj/effect/proc_holder/spell/targeted/sintouch(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/conjure_item/summon_pitchfork/greater(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/fireball/hellish(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/infernal_jaunt(null))
+	owner.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/sintouch(null))
 
-/datum/devilinfo/proc/give_arch_spells()
+/datum/antagonist/devil/proc/give_arch_spells()
```

Comment:
You should have all of these procs check if the mind exists, or you cloud possibly get runtimes. It's not *likely* but it's best to be safe.

It's also a good idea to have can_be_owned() return FALSE if there's no mind, if mostly as a concession to horrible antag code.